### PR TITLE
inject option function support

### PR DESCRIPTION
**Pull request by shirotech**
This gives user full control over how the styles are injected. @egoist can you please take a look, would be awesome if this can be merged. Thanks.

### CODE REVIEW

**Comment by shirotech** [comment]
Not sure why `circleci` can't install `node-sass` for some reason, tests run fine on my machine.

**Comment by cdashrowe** [comment]
> Not sure why `circleci` can't install `node-sass` for some reason, tests run fine on my machine.

Seems to be an issue with the updated version of node CircleCI is using, having the same issue on my PR.

**Comment by shirotech** [comment]
@cdashrowe I've changed node to use `lts` instead of `latest` which should be more stable and seems to have fixed the issue.

Would be good if anyone has merging rights to look at this. Many thanks.

File: README.md [hunk 1]
```diff
@@ -101,13 +101,15 @@ PostCSS Plugins.
 
 ### inject
 
-Type: `boolean` `object`<br>
+Type: `boolean` `object` `function(file): string`<br>
```
Comment (Himself65) [inline README.md]: what is argument `file` used for?

Comment (shirotech) [reply README.md]: The given function will receive the file params, allowing the user to compile or inject their own css using the same name as the input etc.

Comment (shirotech) [reply README.md]: For example HMR support :)

File: src/postcss-loader.js [hunk 2]
```diff
@@ -189,11 +189,15 @@ export default {
       };`
     }
     if (!shouldExtract && shouldInject) {
-      output += `\nimport styleInject from '${styleInjectPath}';\nstyleInject(css${
-        Object.keys(options.inject).length > 0 ?
-          `,${JSON.stringify(options.inject)}` :
-          ''
-      });`
+      if (typeof options.inject === 'function') {
+        output += options.inject(this.id)
```
Comment (Himself65) [inline src/postcss-loader.js]: should we add more parameters in `inject` function?

example:
```ts
function inject(fileID: string, variableName: string, css: string): string;
```

need more discuss

**Comment by Himself65** [comment]
landed at 545dc80 and b364cbe

**Pull request closed without merging**